### ISSUE TITLE
Provider tests shows result as passed even when it did not find contract file

### ISSUE THREAD

**Issue by heerasinghknp** [pact-foundation/pact-js#927]
_Thank you for reporting a bug! We appreciate it very much. Issues are a big input into the priorities for Pact-JS development_

_All italic text in this template is safe to remove before submitting_

_Thanks again!_

### Software versions

_Please provide at least OS and version of pact-js_

- **OS**: MacBook Pro (16-inch, 2021)
- **Consumer Pact library**: Pact JS "version": "10.0.1"
- **Provider Pact library**: _e.g. pact-jvm-provider-maven_2.11 v 3.3.8_
- **Node Version**: `node --version`

### Issue Checklist

Please confirm the following:

- [ ] I have upgraded to the latest
- [ ] I have the read the FAQs in the Readme
- [ ] I have triple checked, that there are **no unhandled promises** in my code and have [read](https://github.com/pact-foundation/pact-js/blob/master/docs/troubleshooting.md#test-fails-when-it-should-pass) the section on intermittent test failures
- [ ] I have set my log level to debug and attached a log file showing the complete request/response cycle
- [ ] For bonus points and virtual high fives, I have created a reproduceable git repository (see below) to illustrate the problem

### Expected behaviour
when you run provider tests and it does not find the contract file, should throw an error and failed the test run.

### Actual behaviour
I am using jest with pact-js and found when I try to run provider tests which has a wrong mapping with consumer contract file or contract file does not exist.
It still shows Test Suites and Tests as passed.

### Steps to reproduce
Just delete the contract files and run the provider tests 

_How can someone else reproduce this bug?_
Just delete the contract files and run the provider tests 

_For extra bonus points and internet karma, provide a repository, gist or reproducible code snippet so that we can test the problem._
Code sample: https://codeshare.io/j0EEQM 

_We recommend forking the project and modifying a relevant example [in the examples folder](https://github.com/pact-foundation/pact-js/blob/master/examples)_

### Relevant log files

_Please ensure you set logging to `DEBUG` and attach any relevant log files here (or link to a gist)._

**Comment by TimothyJones-DD** [comment]
How are you invoking the provider tests?

**Comment by mefellows** [comment]
I checked the other day, it's definitely a bug - if the URL doesn't exist it just carries on.

**Comment by TimothyJones** [comment]
Weird. The opposite problem is reported here https://github.com/pact-foundation/pact-js-core/issues/397 - probably an inconsistency

**Comment by mefellows** [comment]
Yes, because they are two different sources. One is a dynamic check from the broker, and the core is deliberately erroring if nothing is found. In this case, we should have an option to allow users to "not fail if nothing found".

The URL is a direct pact source. That seems like it should just simply error if the file doesn't exist. We have some validation in this module (that is clearly not working correctly), but ultimately that should be checked in the core.

**Comment by TimothyJones** [comment]
Ahh, makes sense.
